### DISCUSSION TITLE
Does HSSM support Ornstein–Uhlenbeck model with between trial variability parameters?

### DISCUSSION THREAD

**Discussion by Jiashun97** [lnccbrown/HSSM#268] [General]
Hello,

I was wondering how I can specify an Ornstein–Uhlenbeck model with between trial variability parameters. I saw how to add a full DDM model using blackbox likelihoods. Can I also add another parameter 'g' in the full DDM to do the work?

Also, could you give me a pointer to how to estimate parameters for different conditions? I know how to do this things in HDDM, but it seems like HDDM does not support Ornstein–Uhlenbeck model? 

Thanks for your help!

Best regards,
Jiashun

**Comment by digicosmos86** [comment]
Hi Jiashun,

We are adding support for full DDM likelihood support via blackbox in our next release which is coming in the next few weeks. Please stay tuned! @AlexanderFengler might know more about the Ornstein-Uhlenbeck model.

Thanks!
Paul

**Reply by Jiashun97** [reply]
Hi Paul,

Thanks for your quick reply. I am also wondering wondering if it is possible to estimate parameters for multiple conditions at the same time. For example, there is a keyword in hddm.HDDM() called `depends_on`, in which I can specify which conditions the parameter depends on. Does hssm also have similar feature?

Best regards,
Jiashun

**Reply by AlexanderFengler** [reply]
In HSSM the idea is to specify the `depends_on` functionality via the regression functions. 
E.g. if you have a categorical variable `cat_var` and you think `v` (let's say context is the DDM) should depend on this variable, you can specify a regression function for `v` such as:

`v ~ 1 + cat_var` 

This is essentially the same thing you would have done in HDDM with the HDDMRegressor class, through which you were able to replicate the `depends_on` behavior. 

In HDDM there would have been some computational advantages to using `depends_on` in specific cases, but this doesn't translate to how computations are executed in HSSM.

Hope this helps.

**Reply by Jiashun97** [reply]
Thanks a lot for your detailed explanation!

**Comment by frankmj** [comment]
See this thread for how to fit by condition
https://github.com/lnccbrown/HSSM/issues/224

For O-U there is currently no between trial variability but one can train a
new LAN to try that - though there may be issues with recovery


Michael

**Reply by Jiashun97** [reply]
Thanks for your suggestion! I will give it a try.
